plugins/catalog-backend-module-openapi-spec/src/InternalOpenApiDocumentationProvider.ts

```diff
@@ -0,0 +1,244 @@
+      logger.error(
+        `Failed to load spec for plugin=${pluginId} at ${openApiUrl}. Error (${
+          response.status
+        }): ${response.body ? await response.text() : response.statusText}`,
+      );
+    }
+  }
+  return mergeSpecs({ baseUrl, specs });
+};
+
+export class InternalOpenApiDocumentationProvider implements EntityProvider {
+  private connection?: EntityProviderConnection;
+  private readonly scheduleFn: () => Promise<void>;
+  constructor(
+    public readonly config: Config,
+    public readonly discovery: DiscoveryService,
+    public readonly logger: LoggerService,
+    taskRunner: TaskRunner,
+  ) {
+    this.scheduleFn = this.createScheduleFn(taskRunner);
+  }
+
+  static fromConfig(
+    config: Config,
+    options: {
+      discovery: DiscoveryService;
+      logger: LoggerService;
+      schedule: PluginTaskScheduler;
+    },
+  ) {
+    const taskRunner = options.schedule.createScheduledTaskRunner({
+      frequency: {
+        minutes: 1,
+      },
+      timeout: {
+        minutes: 1,
+      },
+    });
+    return new InternalOpenApiDocumentationProvider(
+      config,
+      options.discovery,
+      options.logger,
+      taskRunner,
+    );
+  }
+  /** {@inheritdoc @backstage/plugin-catalog-backend#EntityProvider.getProviderName} */
+  getProviderName() {
+    return `InternalOpenApiDocumentationProvider`;
+  }
+
+  /** {@inheritdoc @backstage/plugin-catalog-backend#EntityProvider.connect} */
+  async connect(connection: EntityProviderConnection) {
+    this.connection = connection;
+    return await this.scheduleFn();
+  }
+
+  private createScheduleFn(taskRunner: TaskRunner): () => Promise<void> {
+    return async () => {
+      const taskId = `${this.getProviderName()}:refresh`;
+      return taskRunner.run({
+        id: taskId,
+        fn: async () => {
+          const logger = this.logger.child({
+            class:
+              InternalOpenApiDocumentationProvider.prototype.constructor.name,
+            taskId,
+            taskInstanceId: uuid.v4(),
+          });
+          try {
+            await this.refresh(logger);
+          } catch (error) {
+            logger.error(`${this.getProviderName()} refresh failed`, error);
+          }
+        },
+      });
+    };
+  }
+
+  async refresh(logger: LoggerService) {
+    const pluginsToMerge = this.config.getStringArray('openapi.plugins');
+    logger.info(`Loading specs from from ${pluginsToMerge}.`);
+    const documentationEntity: ApiEntity = {
+      apiVersion: 'backstage.io/v1beta1',
+      kind: 'API',
+      metadata: {
+        name: 'INTERNAL_instance_openapi_doc',
+        title: 'Your Backstage Instance documentation',
+        annotations: {
+          'backstage.io/managed-by-location':
+            'internal-package:@backstage/plugin-catalog-backend-module-openapi-spec',
+          'backstage.io/managed-by-origin-location':
+            'internal-package:@backstage/plugin-catalog-backend-module-openapi-spec',
```

Comment:
`ANNOTATION_LOCATION` and `ANNOTATION_ORIGIN_LOCATION` for these.

I take it this entire structure is fairly experimental and likely to evolve? Otherwise I'm thinking it would make sense to make it possible to configure the name and perhaps some other fields in the entity.

Reply:
Yeah, for now I'd like to keep the extensability small as this is all _very_ experimental